### PR TITLE
[WebNN EP] WebNN only supports 4-D input and weight for Conv/ConvTranspose

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
image

Whisper tiny encoder model contains two conv nodes that have only 3-D inputs. Currently WebNN only supports 4-D inputs. So just fallback them. 

@fdwr, @guschmue, PTAL, thanks!

**Comment by Honry** [comment]
@fdwr, shall we support 3-D inputs by prepending a 1 size dimension?

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc [hunk 1]
```diff
@@ -293,22 +293,39 @@ bool ConvOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
...
+  }
+
+  const auto weight_size = weight_shape.size();
+  if (weight_size != 4) {
+    LOGS(logger, VERBOSE) << op_type << " [" << name << "]'s weight dimension: " << weight_size
+                          << ". Only conv 2d is supported.";
+    return false;
+  }
+
   // WebNN CPU backend (XNNPACK) requires the filter operand to be a constant.
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:322]: I hope XNNPack can fix this in the future...

**Review by fdwr — APPROVED** [review]
👍

**Comment by fdwr** [comment]
> shall we support 3-D inputs by prepending a 1 size dimension?

Yes, we should support 1D to 4D like the DML EP by reshaping with prepended 1's (obviously not 5D because WebNN doesn't have support).

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:322]: I hope so. :)

**Comment by Honry** [comment]
> Yes, we should support 1D to 4D like the DML EP by reshaping with prepended 1's (obviously not 5D because WebNN doesn't have support).

Agree, I will support that in a follow up CR.

**Pull request merged**